### ISSUE TITLE
Propagate comments from parser to generated rust code

### ISSUE THREAD

**Issue by tohrnii** [0xPolygonMiden/air-script#66]
Currently comments written in the AirScript file are lost after parsing. We should preserve the comments and output them at the relevant places in the generated rust code.

**Comment by bobbinth** [comment]
This might be really tricky especially when we start optimizing constraint expressions and proving a more module way of describing complex constraints.

**Comment by tohrnii** [comment]
> This might be really tricky especially when we start optimizing constraint expressions and proving a more module way of describing complex constraints.

Makes sense. Maybe we can limit the type of comments that gets propagated? For example comments for things like constant definitions should be easy to propagate. Or if there are general comments pertaining to a whole section. If we go with this kind of restrictive kind of comments, maybe we can have a specialized syntax for comments that will get propagated and we can restrict where in the AIR file they can be used while parsing?

**Comment by bobbinth** [comment]
Yes, I think we can come up with some good approaches here - but I'd wait until we have much more of the language implemented before tackling this (probably after we have support for spreading constraint descriptions over multiple modules/files).

**Comment by tohrnii** [comment]
Another option for this specific issue is to limit the scope of the issue to just preserve the comments in the parser for now (what @grjte originally suggested).
